compositor/hmd.cc

```diff
@@ -88,20 +90,25 @@ Matrix4 HMD::ConvertSteamVRMatrixToMatrix(vr::HmdMatrix34_t &pose)
   );
 }
 
-Matrix4 HMD::ProjectionMatrix()
+Matrix4 HMD::ProjectionMatrix(vr::Hmd_Eye hmd_eye)
 {
   if (!vr_system_) return Matrix4();
 
   float far = 1000.0f;
   float near = 0.1f;
-  float e = -2 * (far * near) / (far - near);
-  float f = (far + near) / (far - near);
-
-  return Matrix4(  //
-      1, 0, 0, 0,  //
-      0, 1, 0, 0,  //
-      0, 0, f, 1,  //
-      0, 0, e, 0   //
+  float left, right, bottom, top;
+  vr_system_->GetProjectionRaw(hmd_eye, &left, &right, &top, &bottom);
+  float dx = 1.0f / (right - left);
+  float dy = 1.0f / (bottom - top);
+  float dz = 1.0f / (far - near);
+  float sx = right + left;
+  float sy = bottom + top;
+
+  return Matrix4(                                   //
+      2 * dx, 0, 0, 0,                              //
+      0, 2 * dy, 0, 0,                              //
+      sx * dx, sy * dy, -(far + near) * dz, -1.0f,  //
+      0, 0, -2 * far * near * dz, 0                 //
```

Comment:
```suggestion
  return Matrix4(                                   //
      2 * dx, 0, 0, 0,                              //
      0, 2 * dy, 0, 0,                              //
      -sx * dx, -sy * dy, (far + near) * dz, 1.0f,  //
      0, 0, -far * near * dz, 0                 //
```

![IMG_C4E49174E8AC-1](https://user-images.githubusercontent.com/44729662/127778617-e124597e-b4df-4f44-9210-d8dd634e9349.jpeg)

Reply:
コレもちろん試したんだけど上手く行かないんよね（HMDが向いた方向に3Dモデルがついてきてしまう + 世界座標系もz軸がマイナスが正面のまま）
視錐台のz軸を反転させても上手く行かなさそう
View Matrixの向きも合ってそうなんだよね。。もうちょい調べてみる。

ちなみに 2* にしてるのは床井さんの記事の内容を参考にしてる。どっちでも良さそう。
https://marina.sys.wakayama-u.ac.jp/~tokoi/?date=20090829

Reply:
2* するのはz軸をNDCのz軸の[-1, 1]に写像する時で、[0, 1]に写像する時は2*しないみたいだね
OpenGLはデフォルトでNDCの[-1, 1]をクリッピングするみたいだから2*が必要そうでした。
2*で、符号を反転させたのも試した？

Reply:
それも試したけどダメだった。
何も描画されない。